### PR TITLE
Fixes VSTS Bug 1042154: Wrong accessibility translations

### PR DESCRIPTION
**Pull request by mkrueger**
https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1042154

### CODE REVIEW

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/EditorCompareWidgetBase.MiddleArea.cs [hunk 1]
```diff
@@ -480,14 +480,13 @@ public ButtonAccessible (MiddleArea widget, Hunk hunk)
...
 
 					if (hunk.Inserted > 0) {
-						Accessible.Label = GettextCatalog.GetString ("Revert {0} inserted lines starting at {1}", hunk.Inserted, hunk.InsertStart);
+						Accessible.Label = string.Format (GettextCatalog.GetPluralString ("Revert {0} inserted line starting at {1}", "Revert {0} inserted lines starting at {1}", hunk.Inserted), hunk.InsertStart);
 					} else if (hunk.Removed > 0) {
-						Accessible.Label = GettextCatalog.GetString ("Revert {0} removed lines starting at {1}", hunk.Removed, hunk.RemoveStart);
+						Accessible.Label = string.Format (GettextCatalog.GetString ("Revert {0} removed line starting at {1}", "Revert {0} removed lines starting at {1}", hunk.Removed), hunk.RemoveStart);
 					} else {
-						Accessible.Label = GettextCatalog.GetString ("Revert {0} replaced lines starting at {1}", hunk.Removed, hunk.InsertStart);
+						Accessible.Label = string.Format (GettextCatalog.GetString ("Revert {0} replaced line starting at {1}", "Revert {0} replaced lines starting at {1}", hunk.Removed), hunk.InsertStart);
```
Comment (mrward) [inline main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/EditorCompareWidgetBase.MiddleArea.cs]: Some of these are using GetPluralString whilst some are not.

Also do we need to use String.Format? I thought GetPluralString would handle this for you? CustomToolService uses GetPluralString with more than one string.format parameter:

https://github.com/mono/monodevelop/blob/ef02639623d59e6a32c36c7c3caa77637f82267e/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.CustomTools/CustomToolService.cs#L238-L242

Comment (mkrueger) [reply main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/EditorCompareWidgetBase.MiddleArea.cs]: Didn't know that getpluralstring has overloads for a format. /fixed

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/EditorCompareWidgetBase.MiddleArea.cs [hunk 2]
```diff
@@ -480,14 +480,13 @@ public ButtonAccessible (MiddleArea widget, Hunk hunk)
...
 
 					if (hunk.Inserted > 0) {
-						Accessible.Label = GettextCatalog.GetString ("Revert {0} inserted lines starting at {1}", hunk.Inserted, hunk.InsertStart);
+						Accessible.Label = GettextCatalog.GetPluralString ("Revert {0} inserted line starting at {1}", "Revert {0} inserted lines starting at {1}", hunk.Inserted, hunk.InsertStart);
 					} else if (hunk.Removed > 0) {
-						Accessible.Label = GettextCatalog.GetString ("Revert {0} removed lines starting at {1}", hunk.Removed, hunk.RemoveStart);
+						Accessible.Label = GettextCatalog.GetPluralString ("Revert {0} removed line starting at {1}", "Revert {0} removed lines starting at {1}", hunk.Removed, hunk.RemoveStart);
 					} else {
-						Accessible.Label = GettextCatalog.GetString ("Revert {0} replaced lines starting at {1}", hunk.Removed, hunk.InsertStart);
+						Accessible.Label = GettextCatalog.GetPluralString ("Revert {0} replaced line starting at {1}", "Revert {0} replaced lines starting at {1}", hunk.Removed, hunk.InsertStart);
```
Comment (mrward) [inline main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/EditorCompareWidgetBase.MiddleArea.cs]: Not sure this looks correct to me. GetPluralString needs a parameter to indicate which string to use, then you need to pass the two (0, and 1) arguments. So shouldn't there be 3 parameters here?

We should be calling this overload of GetPluralString:

https://github.com/mono/monodevelop/blob/d4a8ca63d1c0c906605810fb067409cf1b00185e/main/src/core/MonoDevelop.Core/MonoDevelop.Core/Gettext.cs#L221

But it looks like we are calling this other one:

https://github.com/mono/monodevelop/blob/d4a8ca63d1c0c906605810fb067409cf1b00185e/main/src/core/MonoDevelop.Core/MonoDevelop.Core/Gettext.cs#L216

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/EditorCompareWidgetBase.MiddleArea.cs [hunk 3]
```diff
@@ -480,14 +480,13 @@ public ButtonAccessible (MiddleArea widget, Hunk hunk)
 					Accessible.PerformPress += PerformPress;
 
 					Accessible.SetRole (AtkCocoa.Roles.AXButton);
-					Accessible.Label = widget.useLeft ? GettextCatalog.GetString ("Revert left changes button") : GettextCatalog.GetString ("Revert right changes button");
 
 					if (hunk.Inserted > 0) {
-						Accessible.Label = GettextCatalog.GetString ("Revert {0} inserted lines starting at {1}", hunk.Inserted, hunk.InsertStart);
+						Accessible.Label = GettextCatalog.GetPluralString ("Revert {0} inserted line starting at {1}", "Revert {0} inserted lines starting at {1}", hunk.Inserted, hunk.InsertStart);
```
Comment (sevoku) [inline main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/EditorCompareWidgetBase.MiddleArea.cs]:
```suggestion
						Accessible.Label = GettextCatalog.GetPluralString ("Revert {0} inserted line starting at {1}", "Revert {0} inserted lines starting at {1}", hunk.Inserted, hunk.Inserted, hunk.InsertStart);
```

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/EditorCompareWidgetBase.MiddleArea.cs [hunk 4]
```diff
@@ -480,14 +480,13 @@ public ButtonAccessible (MiddleArea widget, Hunk hunk)
...
 					} else if (hunk.Removed > 0) {
-						Accessible.Label = GettextCatalog.GetString ("Revert {0} removed lines starting at {1}", hunk.Removed, hunk.RemoveStart);
+						Accessible.Label = GettextCatalog.GetPluralString ("Revert {0} removed line starting at {1}", "Revert {0} removed lines starting at {1}", hunk.Removed, hunk.RemoveStart);
```
Comment (sevoku) [inline main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/EditorCompareWidgetBase.MiddleArea.cs]:
```suggestion
						Accessible.Label = GettextCatalog.GetPluralString ("Revert {0} removed line starting at {1}", "Revert {0} removed lines starting at {1}", hunk.Removed, hunk.Removed, hunk.RemoveStart);
```

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/EditorCompareWidgetBase.MiddleArea.cs (hunk 2 above)
Comment (sevoku) [inline main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/EditorCompareWidgetBase.MiddleArea.cs]:
```suggestion
						Accessible.Label = GettextCatalog.GetPluralString ("Revert {0} replaced line starting at {1}", "Revert {0} replaced lines starting at {1}", hunk.Removed, hunk.Removed, hunk.InsertStart);
```

Comment (mkrueger) [reply main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/EditorCompareWidgetBase.MiddleArea.cs]: Seems I did that wrong all these years. Thought that the arg0 is always used to determine the string.

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/EditorCompareWidgetBase.MiddleArea.cs (hunk 3 above)
Comment (mkrueger) [reply main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/EditorCompareWidgetBase.MiddleArea.cs]: fixed.

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/EditorCompareWidgetBase.MiddleArea.cs (hunk 2 above)
Comment (mkrueger) [reply main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/EditorCompareWidgetBase.MiddleArea.cs]: fixed.

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/EditorCompareWidgetBase.MiddleArea.cs (hunk 4 above)
Comment (mkrueger) [reply main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/EditorCompareWidgetBase.MiddleArea.cs]: fixed.

**Comment by sevoku** [comment]
@monojenkins rebase

**Comment by sevoku** [comment]
@monojenkins backport release-8.4-a11y

**Comment by sevoku** [comment]
@monojenkins backport release-8.4-a11y

**Pull request merged**